rfqmath/fixed_point_test.go

```diff
@@ -159,6 +159,144 @@ func testFromUint64[N Int[N]](t *rapid.T) {
 	require.Equal(t, coefficient, scaledBack.Coefficient.ToUint64())
 }
 
+// TestWithinTolerance tests that the FixedPoint.WithinTolerance method works as
+// expected.
+func testWithinTolerance[N Int[N]](t *rapid.T) {
+	type testCase struct {
+		// firstFp is the fixed-point to compare with secondFp.
+		firstFp FixedPoint[N]
+
+		// secondFp is the fixed-point to compare with firstFp.
+		secondFp FixedPoint[N]
+
+		// tolerancePpm is the tolerance in parts per million (PPM) that
+		// the second price can deviate from the first price and still
+		// be considered within bounds.
+		tolerancePpm uint64
+
+		// withinBounds is the expected result of the bounds check.
+		withinBounds bool
+	}
+
+	testCases := []testCase{
+		{
+			// Case where secondFp is 10% less than firstFp,
+			// tolerance allows 11.11% (111111 PPM). Diff within
+			// bounds.
+			firstFp:      FixedPointFromUint64[N](1000000, 1),
+			secondFp:     FixedPointFromUint64[N](900000, 1),
```

Comment:
Are there any overflow cases to handle?
Seems there should be sufficient space:
>log(2**64)/log(1000000*900000) = 1.6

Reply:
`WithinTolerance` uses methods from the `rfqmath.Arithmetic` interface for calculations, so I think the overflow responsibility should be on those methods. I'm not sure if its possible to get the `Mul` calls to overflow.

Reply:
> I'm not sure if its possible to get the Mul calls to overflow.

So we'll use the `BigInt[T]` version everywhere, which for all practical purposes, can't actually overflow.